### DISCUSSION TITLE
Improve responsive design for smaller screen sizes with a breakpoint provider component?

### DISCUSSION THREAD

**Discussion by PascalVorwerk** [microsoft/fluentui-blazor#2895] [Ideas]
Hi! I was wondering if there is any interest to enhance the development for smaller screen sizes in Fluent UI through things like break point providers or css utility classes. Some other providers of components offer these things:

Mudblazor: https://mudblazor.com/components/breakpointprovider#api
Syncfusion: https://blazor.syncfusion.com/documentation/media-query/break-points

(And probably a few more)

I think it would be interesting to build one of these approaches into this library to enhance the capabilities of developers to improve their UI for smaller screens. I know there already is a Grid component that allows you to do this, but this is more about making it easier to 'change UI depening on screen sizes'. For example, it might make it easier to collapse a navmenu depending on breakpoints.

**Comment by dvoituron** [comment]
Like you said, you already have the [FluentGrid](https://www.fluentui-blazor.net/Grid) and nothing else is planned at this moment.

What do you want and what is not possible with the **FluentGrid**, because you have `xs`, `sm`, ... attributes but also the `HiddenWhen` attribute which seems similar to the features included in your links. Your have also `OnBreakpointEnter` to detect when the break point changes.

**Reply by PascalVorwerk** [reply]
Well for certain things it would be nice to be able to query the media, which ofcourse you could do through javascript interop on its own, but i just see these ways of getting the current 'Breakpoint' in other libraries, which I imagine could be handy?

Like i said: The collapsible toggle on a navmenu would be easier to implement based on current window size / breakpoint if there was some kind of provider right? Or is there a way to do this with simply the grid that I failed to understand?

**Reply by dvoituron** [reply]
You can do this with the FluentGrid:
1. Use the `HiddenWhen` attribute to hide some DIV when you want (if you have DIV to show/hide in your menu)
2. Use the `OnBreakpointEnter` to catch the media size in a local variable. And use it with your NavMenu.

**Reply by PascalVorwerk** [reply]
Right, but to me it does kinda seem like the Grid then becomes a 'breakpoint' provider instead of a grid system. Let's say i wrap my navmenu in a grid and i simply would want to toggle the menu to collapse when screen size is <md, the grid component would only be there for me to listen to those changes and set the boolean value to true/false depending on those changes?

Could that be the reason why for example mudblazor provides both a breakpoint provider AND a grid system like we have here?

**Reply by dvoituron** [reply]
Yes. For us, this component manages the both 🙂

**Reply by fdonnet** [reply]
I try to always use the standard components. But like you, to not be too nervous, I go with Tailwind (with tw- prefix for class) sometimes. Ex for flexible div (FluentSack equivalent), I go with "sm:" or other Tailwind breakpoints to change the orientation of the div and not use FluentStack.

Tried with fluentgrid but when you don t want "col" but px or rem size for one of the element. It begins to be hard.

Dont get me wrong, the work on components is awesome. But I miss the Tailwind way a lot as a backend guy who doesn t love CSS or theming.

The Tailwind, breakpoints styling, is just so good..you begin with mobile and you add your change after for bigger screens. And it s hard to loose that and use tricks in code or even additional CSS. ;)

**Comment by Tyme-Bleyaert** [comment]
I also have use the `FluentDataGrid` and wanted to have a card like grid on mobile versions. So what i ended up doing was write a `ResponsiveContent` [component](https://github.com/TymeB-Examples/ResponsiveRenderingBlazor) wrapper and based on the breakpoint context i show more or less columns and change the class.
